### PR TITLE
Implements targeting and attacking functionality using middlewares

### PR DESCRIPTION
**Pull request by brunosabenca**

### CODE REVIEW

File: src/ducks/reducer.js [hunk 1]
```diff
@@ -276,7 +248,9 @@ export default function reducer(state = initialState, action = {}) {
               name,
               cost,
               description,
-              actions
+              actions,
+              needsTarget,
+              target
```
Comment (L3v3L) [inline src/ducks/reducer.js]: why is target in the ref? this ref is used by all cards with the same ref. if you place something here all of them will have the same target.

Comment (brunosabenca) [reply src/ducks/reducer.js]: It's a mistake, it's not needed.

File: src/components/Card.js [hunk 2]
```diff
@@ -39,7 +35,7 @@ const CardButton = styled.div`
     1px 1px 0 #000;
   .interiorContainer {
     user-select: none;
-    border: ${props => props.cardValues.innerBorder}px solid #2f3534;
+    border: ${props => props.cardValues.innerBorder}px solid ${props => props.isActive ? 'red' : '#2f3534'};
```
Comment (L3v3L) [inline src/components/Card.js]: don't think this works

Comment (brunosabenca) [reply src/components/Card.js]: Nope, forgot to remove it.

**Pull request merged**